<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>738</width>
    <height>466</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="CodeEditor" name="plainTextEdit"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>738</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_F">
    <property name="title">
     <string>文件(&amp;F)</string>
    </property>
    <addaction name="action_N"/>
    <addaction name="action_O"/>
    <addaction name="action_S"/>
    <addaction name="action_A"/>
    <addaction name="separator"/>
    <addaction name="action_X"/>
   </widget>
   <widget class="QMenu" name="menu_E">
    <property name="title">
     <string>编辑(&amp;E)</string>
    </property>
    <addaction name="action_U"/>
    <addaction name="action_Z"/>
    <addaction name="action_T"/>
    <addaction name="action_C"/>
    <addaction name="action_P"/>
    <addaction name="separator"/>
    <addaction name="action_F"/>
    <addaction name="action_R"/>
    <addaction name="separator"/>
    <addaction name="action_All"/>
   </widget>
   <widget class="QMenu" name="menu_O">
    <property name="title">
     <string>格式(&amp;O)</string>
    </property>
    <addaction name="action_W"/>
    <addaction name="action_L"/>
    <addaction name="action_Font"/>
    <addaction name="action_Color"/>
    <addaction name="separator"/>
    <addaction name="action_FontBackgroundColor"/>
    <addaction name="action_EditorBackgroundColor"/>
   </widget>
   <widget class="QMenu" name="menu_V">
    <property name="title">
     <string>查看(&amp;V)</string>
    </property>
    <addaction name="action_Tool"/>
    <addaction name="action_Statu"/>
   </widget>
   <widget class="QMenu" name="menu_H">
    <property name="title">
     <string>帮助(&amp;H)</string>
    </property>
    <addaction name="action_About"/>
   </widget>
   <addaction name="menu_F"/>
   <addaction name="menu_E"/>
   <addaction name="menu_O"/>
   <addaction name="menu_V"/>
   <addaction name="menu_H"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="styleSheet">
    <string notr="true">/* 工具栏整体样式 */
QToolBar {
    background-color: #f0f0f0; /* 工具栏背景色 */
    border: 1px solid #d0d0d0; /* 边框 */
    border-radius: 4px; /* 圆角 */
    padding: 1px; /* 内边距 */
    spacing: 1px; /* 按钮间距 */
}

/* 工具栏按钮基础样式 */
QToolBar QToolButton {
    background-color: transparent; /* 透明背景 */
    border: 1px solid transparent; /* 透明边框 */
    border-radius: 3px; /* 圆角 */
    padding: 4px 6px; /* 内边距 */
    color: #333333; /* 文字颜色 */
}

/* 按钮悬停状态 */
QToolBar QToolButton:hover {
    background-color: #e0e0e0; /* 悬停背景色 */
    border-color: #c0c0c0; /* 悬停边框色 */
}

/* 按钮按下状态 */
QToolBar QToolButton:pressed {
    background-color: #d0d0d0; /* 按下背景色 */
    border-color: #b0b0b0; /* 按下边框色 */
}

/* 按钮选中状态（如切换按钮） */
QToolBar QToolButton:checked {
    background-color: #d5e8ff; /* 选中背景色 */
    border-color: #99c4ff; /* 选中边框色 */
}

/* 禁用状态按钮 */
QToolBar QToolButton:disabled {
    color: #aaaaaa; /* 禁用文字色 */
}

/* 工具栏分隔线 */
QToolBar QFrame[frameShape=&quot;QFrame::VLine&quot;] {
    background-color: #d0d0d0; /* 分隔线颜色 */
    width: 1px; /* 分隔线宽度 */
    margin: 4px 2px; /* 分隔线边距 */
}

/* 带菜单的按钮指示器 */
QToolBar QToolButton::menu-indicator {
    image: url(:/icons/menu_indicator.png); /* 自定义指示器图标 */
    subcontrol-position: right center; /* 位置 */
    subcontrol-origin: padding; /* 原点 */
    width: 8px; /* 宽度 */
    height: 8px; /* 高度 */
    right: 2px; /* 右边距 */
}</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_N"/>
   <addaction name="action_O"/>
   <addaction name="action_S"/>
   <addaction name="action_A"/>
   <addaction name="separator"/>
   <addaction name="action_U"/>
   <addaction name="action_Z"/>
   <addaction name="action_T"/>
   <addaction name="action_C"/>
   <addaction name="action_P"/>
   <addaction name="separator"/>
   <addaction name="action_F"/>
   <addaction name="action_R"/>
   <addaction name="separator"/>
   <addaction name="action_Font"/>
   <addaction name="action_L"/>
   <addaction name="action_W"/>
   <addaction name="separator"/>
   <addaction name="action_Tool"/>
   <addaction name="action_Statu"/>
  </widget>
  <action name="action_N">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentNew"/>
   </property>
   <property name="text">
    <string>新建(&amp;N)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_O">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentOpen"/>
   </property>
   <property name="text">
    <string>打开(&amp;O)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_S">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentSave"/>
   </property>
   <property name="text">
    <string>保存(&amp;S)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_A">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentSaveAs"/>
   </property>
   <property name="text">
    <string>另存为(&amp;A)</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_X">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::ApplicationExit"/>
   </property>
   <property name="text">
    <string>退出(&amp;X)</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_U">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditUndo"/>
   </property>
   <property name="text">
    <string>撤销(&amp;U)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_Z">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditRedo"/>
   </property>
   <property name="text">
    <string>恢复(&amp;Z)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_T">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditCut"/>
   </property>
   <property name="text">
    <string>剪切(&amp;T)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_C">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditCopy"/>
   </property>
   <property name="text">
    <string>复制(&amp;C)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_P">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditPaste"/>
   </property>
   <property name="text">
    <string>粘贴(&amp;P)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_F">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditFind"/>
   </property>
   <property name="text">
    <string>查找(&amp;F)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_R">
   <property name="icon">
    <iconset theme="accessories-character-map"/>
   </property>
   <property name="text">
    <string>替换(&amp;R)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_All">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditSelectAll"/>
   </property>
   <property name="text">
    <string>全选(&amp;A)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_W">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/img/自动换行.png</normaloff>:/img/自动换行.png</iconset>
   </property>
   <property name="text">
    <string>自动换行(&amp;W)</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_L">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/img/显示行号.png</normaloff>
     <normalon>img/显示行号.png</normalon>:/img/显示行号.png</iconset>
   </property>
   <property name="text">
    <string>显示行号(&amp;L)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_Font">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/img/字体.png</normaloff>:/img/字体.png</iconset>
   </property>
   <property name="text">
    <string>字体(&amp;F)</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_Color">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/img/字体颜色_文字操作_jurassic.png</normaloff>:/img/字体颜色_文字操作_jurassic.png</iconset>
   </property>
   <property name="text">
    <string>字体颜色</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_FontBackgroundColor">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/img/字体背景颜色.png</normaloff>:/img/字体背景颜色.png</iconset>
   </property>
   <property name="text">
    <string>字体背景颜色</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_EditorBackgroundColor">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/img/背景颜色.png</normaloff>:/img/背景颜色.png</iconset>
   </property>
   <property name="text">
    <string>编辑器背景色</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_Tool">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentProperties"/>
   </property>
   <property name="text">
    <string>工具栏(&amp;T)</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_Statu">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/img/3状态栏.png</normaloff>:/img/3状态栏.png</iconset>
   </property>
   <property name="text">
    <string>状态栏(&amp;S)</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_About">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::HelpAbout"/>
   </property>
   <property name="text">
    <string>关于(&amp;A)</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">codeeditor.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
